### ISSUE TITLE
Can't open app

### ISSUE THREAD

**Issue by alr86** [subhra74/xdm#1130]
**Describe the bug**
The app will not open
In fact, when I want to download from the browser, XDM opens and the download is successfully completed, but the main user interface itself does not open.

**Information:**
 - OS: Arch Linux - KDE Plasma
 - Browser: Firefox
 - XDM addon Version: 3..4
 - XDM Version: 8.0.29 BETA

**Terminal result:**

> 
>  I  ~  xdman
> [xdm-10:45:23] Loading config...
> [xdm-10:45:24] Application_Startup
> 
> (xdm-app:27785): Gtk-WARNING **: 10:45:24.285: Locale not supported by C library.
> 	Using the fallback 'C' locale.
> [xdm-10:45:24] Loading languages...
> [xdm-10:45:24] Language loading ...
> [xdm-10:45:24] Loading lang files from: /opt/xdman/Lang/English.txt
> [xdm-10:45:24] Loading lang files from: /opt/xdman/Lang/English.txt
> [xdm-10:45:24] Language loaded.
> 
> (xdm-app:27785): Gtk-CRITICAL **: 10:45:25.042: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
> /usr/bin/xdman: line 3: 27785 Segmentation fault      (core dumped) /opt/xdman/xdm-app $@
>

**Comment by vhbui02** [comment]
@alr86 I'm having the same issue, it was running fine in the 1st and 2nd run, but no longer after the 3rd :v

**Comment by Proj-Null** [comment]
got the same issue on Debian 12(kde).
when i try to launch i get this.
```[xdm-17:51:06] Loading config...
[xdm-17:51:06] Application_Startup
[xdm-17:51:07] Loading languages...
[xdm-17:51:07] Language loading ...
[xdm-17:51:07] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-17:51:07] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-17:51:07] Language loaded.```

(xdm-app:42174): Gtk-CRITICAL **: 17:51:07.317: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
/usr/bin/xdman: line 3: 42174 Segmentation fault      (core dumped) /opt/xdman/xdm-app $@

**Comment by telmob** [comment]
I have the same problem. It only opens as root.

**Comment by LinuxAndCoffee** [comment]
I have the same issue and have almost the exact same specs as the original poster. Only difference is I am on EndeavourOS (which is still basically Arch). I am running it under Wayland, not X11, but when I tried X11 it still did not work. This is what I get if I try to launch from the terminal:

/opt/xdman/xdm-app                                        
[xdm-13:06:11] Loading config...
[xdm-13:06:11] Application_Startup
[xdm-13:06:11] Loading languages...
[xdm-13:06:11] Language loading ...
[xdm-13:06:11] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-13:06:11] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-13:06:11] Language loaded.

(xdm-app:14430): Gtk-CRITICAL **: 13:06:11.494: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
[1]    14430 segmentation fault (core dumped)  /opt/xdman/xdm-app

**Comment by marifcelik** [comment]
after kill the xdm process, it opens but there is no indicator. I think this is the issue because, I don't know how it happened, for a while the indicator was in the upper right corner and I could open the application without any problems.

**Comment by gaaash** [comment]
same for me
[xdm-04:18:23] Loading config...
[xdm-04:18:24] Application_Startup
Gtk-Message: 04:18:24.322: Not loading module "atk-bridge": The functionality is provided by GTK natively. Please try to not load it.
[xdm-04:18:24] Loading languages...
[xdm-04:18:24] Language loading ...
[xdm-04:18:24] Loading lang files from: /snap/xdman/44/opt/xdman/Lang/English.txt
[xdm-04:18:24] Loading lang files from: /snap/xdman/44/opt/xdman/Lang/English.txt
[xdm-04:18:24] Language loaded.

(xdm-app:7404): Gtk-CRITICAL **: 04:18:25.083: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
Segmentation fault (core dumped)

**Comment by gaaash** [comment]
> after kill the xdm process, it's open but there is no indicator. I think this is the issue because I don't know how it happened but for a while the indicator was in the upper right corner and I could open the application without any problems.

there was a xdm-app process and i killed it with system monitor 
now i can open it again 
thanks

**Comment by vhbui02** [comment]
When UI is closed, xdm isn't terminated, instead it ran in the background. Now the command ```xdman``` in terminal only works if you open it for the 1st time, 2nd time it won't work anymore since there is an xdm process has been initialized before (in my opinion, this is a stupid behavior, why don't it show the UI of the pre xdm?).

Now once xdm went into background, there's nothing I can do anymore since there is no app indicator in top bar.

**Comment by codihuman** [comment]
Yes, this problem covered me too. So I deleted the app by Terminal with root rights `sudo su -`
And used this command:
`yum remove xdman_gtk.x86_64`

And downloaded rpm app by lower version and installed again. Works fine on Fedora 38

**Comment by vhbui02** [comment]
@codihuman which xdman version worked for you? I would like to reinstall as well.

**Comment by codihuman** [comment]
> @codihuman which xdman version worked for you? I would like to reinstall as well.

@Silverbullet069 v. 8.0.26 works fine on my Fedora 38.

**Comment by pcf26536** [comment]
For those on arch or arch based distros, use [aur/xdman8](https://aur.archlinux.org/packages/xdman8)

**Comment by 0ibaba** [comment]
Same here


    OS: Fedora 39 - Gnome
    Browser: Firefox
    XDM addon Version: 3..4
    XDM Version: 8.0.29 BETA


v 8.0.26 works fine but there's still no tray icon.

**Comment by alr86** [comment]
✅ In v8.0.26+ the problem is solved.

**Comment by telmob** [comment]
[xdm-22:53:04] Loading config...
[xdm-22:53:04] Application_Startup
[xdm-22:53:05] Loading languages...
[xdm-22:53:05] Language loading ...
[xdm-22:53:05] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-22:53:05] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-22:53:05] Language loaded.

(xdm-app:25719): Gtk-CRITICAL **: 22:53:05.557: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
/usr/bin/xdman8: line 3: 25719 Segmentation fault      (core dumped) /opt/xdman/xdm-app $@

On Arch xdman8 8.0.29 Beta. Same issue?

**Comment by marifcelik** [comment]
The problem may be due to Wayland. I switched to X11 and I don't have such a problem.

**Comment by betsha1830** [comment]
> > @codihuman which xdman version worked for you? I would like to reinstall as well.
> 
> @Silverbullet069 v. 8.0.26 works fine on my Fedora 38.

Can confirm. V8.0.28 opens in the background but the GUI never appears when closed and opened again. Downloads will be downloaded in the background but there is no way to interact with the main download window. As you've stated, v8.0.26 doesn't have this issue. I'm on Ubuntu 20.04.6 LTS.

**Comment by vhbui02** [comment]
> Can confirm. V8.0.28 opens in the background but the GUI never appears when closed and opened again. Downloads will be downloaded in the background but there is no way to interact with the main download window. As you've stated, v8.0.26 doesn't have this issue. I'm on Ubuntu 20.04.6 LTS.

@betsha1830 I've reinstalled using v8.0.26. I'm also on Ubuntu 22.04 LTS. I've confirmed that the GUI now can be reopened by running `xdman` on terminal or `Alt + F2`. But still, no tray icon. 
This is my speculation: xdman's tray icon has a comparability issue with tray icon extension like `AppIndicator and KStatusNotifierItem Support` or `Ubuntu AppIndicator`

**Comment by Liverel** [comment]
I reported a little while back that there was no system tray indicator showing on Wayland. Subhra needs to implement proper tray support and it will work fine in Wayland, and now lately an even worse issue is happening: the application gets stuck as a background process and can't be brought forward unless you kill the app and start it again. That's not good, hopefully it gets fixed.

**Comment by vhbui02** [comment]
> I reported a little while back that there was no system tray indicator showing on Wayland. Subhra needs to implement proper tray support and it will work fine in Wayland, and now lately an even worse issue is happening: the application gets stuck as a background process and can't be brought forward unless you kill the app and start it again. That's not good, hopefully it gets fixed.

That's true, I'm also using Wayland right now so there isn't any system tray indicator. 
Look at the long list of GitHub Issues that haven't been resolved, I think Subhra and his contributors are pretty busy. I would like to add Wayland support myself but the domain knowledge to understand the underlying structure of xdman is hugh @@

**Comment by fmented** [comment]
this trick worked for me https://github.com/subhra74/xdm/issues/1056#issuecomment-1986459739

**Comment by rinelfi-git** [comment]
Same issue:
```
[xdm-15:03:17] Loading config...
[xdm-15:03:17] Application_Startup
[xdm-15:03:17] Loading languages...
[xdm-15:03:17] Language loading ...
[xdm-15:03:17] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-15:03:17] Loading lang files from: /opt/xdman/Lang/English.txt
[xdm-15:03:17] Language loaded.
/usr/bin/xdman: line 3:  4091 Segmentation fault      /opt/xdman/xdm-app $@
```
I'm using ElementaryOs 7.1 and my XDM version is 8.0.29 BETA

**Comment by zxc2012** [comment]
> this trick worked for me [#1056 (comment)](https://github.com/subhra74/xdm/issues/1056#issuecomment-1986459739)

How to force the app restart in xwayland?

**Comment by fmented** [comment]
> > this trick worked for me [#1056 (comment)](https://github.com/subhra74/xdm/issues/1056#issuecomment-1986459739)
> 
> How to force the app restart in xwayland?

end the xdman-app process, then rerun it with GDK_BACKEND=x11.
or you could modify the desktop entry.

it still doesn't show the system tray menu when i right click on the tray icon, but at least it allows me to open the app from tray icon.

**Comment by JsExpertCoder** [comment]
its open as root, thanks.

> It only opens as root

**Comment by JsExpertCoder** [comment]
its open as root

**Comment by edge07chaos** [comment]
**open it on root mode, as already mentioned**

**Comment by d3cxty** [comment]
its so simple js open as admin
which will be 
sudo xdman

**Comment by fmented** [comment]
forcing an app that doesn't require root access to run as root is a bad practice imo and can be dangerous

**Comment by d3cxty** [comment]
> [@alr86](https://github.com/alr86) I'm having the same issue, it was running fine in the 1st and 2nd run, but no longer after the 3rd :v

always run with sudo
